File: examples-sdk/csharp/029-basic-vil-handler-endpoint/Main.cs
// 029-basic-vil-handler-endpoint — C# SDK equivalent
// Compile: vil compile --from csharp --input 029-basic-vil-handler-endpoint/Main.cs --release

#r "sdk/csharp/Vil.cs"

var server = new VilServer("macro-demo", 8080);
var demo = new ServiceProcess("demo");
demo.Endpoint("GET", "/plain", "plain_handler");
demo.Endpoint("GET", "/handled", "handled_handler");
demo.Endpoint("POST", "/endpoint", "endpoint_handler");
server.Service(demo);
server.Compile();

File: examples-sdk/kotlin/029-basic-vil-handler-endpoint/main.kt
// 029-basic-vil-handler-endpoint — Kotlin SDK equivalent
// Compile: vil compile --from kotlin --input 029-basic-vil-handler-endpoint/main.kt --release

fun main() {
    val server = VilServer("macro-demo", 8080)
    val demo = ServiceProcess("demo")
    demo.endpoint("GET", "/plain", "plain_handler")
    demo.endpoint("GET", "/handled", "handled_handler")
    demo.endpoint("POST", "/endpoint", "endpoint_handler")
    server.service(demo)
    server.compile()
}
